homeassistant/components/aurora_abb_powerone/__init__.py

```diff
@@ -29,9 +32,30 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     comport = entry.data[CONF_PORT]
     address = entry.data[CONF_ADDRESS]
     serclient = AuroraSerialClient(address, comport, parity="N", timeout=1)
-
-    hass.data.setdefault(DOMAIN, {})[entry.unique_id] = serclient
-
+    # To handle yaml import attempts in darkeness, (re)try connecting only if
+    # unique_id not yet assigned.
+    if entry.unique_id is None:
+        try:
+            res = await hass.async_add_executor_job(
+                validate_and_connect, hass, entry.data
+            )
+        except AuroraError as error:
+            if "No response after" in str(error):
```

Comment:
What should we do if this is not the case, but there's an error? We don't want to continue with setup without a unique id.

Reply:
Agreed. Done. 
Now any non-matching error is raised.